homeassistant/helpers/config_validation.py

```diff
@@ -319,7 +319,23 @@ def service(value):
                       .format(value))
 
 
-def slug(value):
+def schema_with_slug_keys(value_schema: Union[T, Callable]) -> Callable:
```

Comment:
This is not needed. Voluptuous supports this out of the box with `vol.Schema({cv.slug: value_schema})`

Reply:
With your approach (and what we currently use), we raise "extra keys not allowed"

```bash
hass --script check_config --config /config
INFO:homeassistant.util.package:Attempting install of colorlog==4.0.2
Testing configuration at /config
Failed config
  homeassistant:
    - extra keys not allowed @ data['packages']['yr__pack']
```

With this PR, and changing the templates, we will raise the suggestion of new slugs, which makes much more sense
```bash
root@linuxkit-00155d5dc21f:/usr/src/app# hass --script check_config --config /config
INFO:homeassistant.util.package:Attempting install of colorlog==4.0.2
Testing configuration at /config
Failed config
  homeassistant:
    - invalid slug yr__pack (try yr_pack) for dictionary value @ data['packages']
```

This is inherent to the design of voluptuous `cv.slug`: The **key** of the template (cv.slug) is not found and then extra keys is raised
[here](https://github.com/alecthomas/voluptuous/issues/127) and [here](https://stackoverflow.com/questions/45847357/getting-a-better-voluptuous-schema-error-message-for-key-type-validation) and no way to fix it there

Open to other suggestions, but we already have [very hacky :-)](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/config.py#L439) error log code to try make this improve this today, without the "this is suggested in the message".

I'm open to suggestions, any other ideas...
```python
cv.All(all_keys_slugs, cv.Schema({str, the value_part})
schema_with_slug_keys(the_value_part)
```